### ISSUE TITLE
Milestone B - load the index page with retracting left menu

### ISSUE THREAD

**Issue by bsaiken** [FWAJL/FieldWorkAssistantMVC#846]
The Field Technician (FT) index, or home page, shall be a list of the FTs assigned tasks.  The Tasks that will appear on this page are where the task is active (task.task_active=1) and the technician.technician_id = task_techncian.techncian_id and task.task_id = task_techncian.task_id.  

If the FT has only one task assigned to them, then skip over this index page and load the task page (Milestone D) directly.  

The index page shall include a breadcrumb in the header with Project Name > Task Name.

Each listed Task on this page shall be a button formatted for a mobile device that includes: 
Deadline (task.task_deadline)
Start date.  This field is not incorporated into the database yet.  You will need to request this field be added to the task table.

Clicking a task or autoloading a task sets the session variable $currentTask and shall have link to the task page (MIlestone D).  If the selected task does not have a start date, include a popup prompt box asking the FT to select their start date.  Use a calendar widget to choose the date.  We currently have datepicker implemented on the site.

**Comment by testingbydev** [comment]
@FWAJL

I am not sure what we have to do in this milestone. can you please help me in identifying the task that i have to do in this milestone. May be i would need your help in 2-3 milestones to understand the way you guys work , Also let me know if you have any user manual for the MVC that we are working upon,

**Comment by FWAJL** [comment]
@testingbydev @bsaiken 

Brian can help you describe what you need to do.

As for the documentation, we have the wiki of the repository and for the rest, please have a look at the code and take one example from the start of the request to the end. 

But before you go further, you need to wait that Milestone A is reviewed by @bsaiken and then @FWAJL.

**Comment by bsaiken** [comment]
@testingbydev 
You must complete Issue #846 before starting on this task.

**Comment by anuj-yadav** [comment]
@bsaiken 
 Please Can you suggest us to resolved some task such as  Adding Keys at "/Resources/Enums"
We didn't get this Point..

**Comment by bsaiken** [comment]
@testingbydev 
You must complete Issue #845 before starting on this task.

**Comment by bsaiken** [comment]
@testingbydev 
Requirements for Milestone B are detailed here:
https://docs.google.com/document/d/1uSf9MTsVg0RzOapusLlkYn2noA2gZXOH86xFuZaZ10s/edit

In order to meet the May 8 deadline, please make initial code submittal by May 6.

**Comment by testingbydev** [comment]
@FWAJL
please tell about the process of how can we assign a project to FieldTechnician.

**Comment by testingbydev** [comment]
@FWAJL
please help us to understanding the Project and Task MVC Code Structure .....

**Comment by testingbydev** [comment]
@FWAJL
We Can't Find Any Session Project In Session if  we have not any project in session then this is redirect Project list to select Project on project Controller but there are User Role 3 not Access the Project Controller So We have to need permitted project  controller for user role 3 if we can't then how can we manage it in Mobile Controller and we cannot understand the process for a project and a task. please help to understanding the file structure and how can we listed task list in mobile controller.

**Comment by bsaiken** [comment]
@testingbydev @FWAJL 

I can answer this question: 
`please tell about the process of how can we assign a project to FieldTechnician.`

You do not need to make any assignments.  The assignments are already made.  

First, understand that Field Technicians are not related to Projects.  For the purpose of your work, the relationship you are interested in is Field Technician <-> Task.  

When a new Field Technician logs in, their identity comes from `user.user_value`.  So, to get the list of Tasks assigned to this User, you would query `user.user_value === task_technician.technician_id`.  The related Task id's are: `task_technician.task_id`.  You will need to restrict this list where `task.task_active===1`.  The are the Tasks that would be listed on the index page.

**Comment by bsaiken** [comment]
@testingbydev @FWAJL 
I can also help a little on this question: 
`We Can't Find Any Session Project In Session if we have not any project in session then this is redirect Project list to select Project on project Controller but there are User Role 3 not Access the Project Controller So We have to need permitted project controller for user role 3 if we can't then how can we manage it in Mobile Controller and we cannot understand the process for a project and a task. please help to understanding the file structure and how can we listed task list in mobile controller.`

You are correct, when a Field Technician logs in, there are no Session Arrays built.  The Session Arrays we have already defined are here: 
https://docs.google.com/spreadsheets/d/1HcS6kgDvSaYjffgWfBO7S_Eq3HKsOYfuL-b-0x5Rczw/edit#gid=1372629556

If that link does not work, go here:
https://github.com/FWAJL/FieldWorkAssistantMVC/wiki/6.-How-to-code-in-the-FWM-code-base
and click on the `Session Array definition document` link. 

You can use our existing code to call any of the predefined arrays.  You will need to identify your Project Manager, so certainly you may need the PM Session array.  We do have a Task Session array; however, it is configured to list Tasks and then the Field Technicians associated with that Task.  You are working with only one Field Technician at a time, so I am not sure if the Task Session array will help you.  You may need to build your own array: FT Session array.  

For any further help, you will need to ask FWAJL.

**Comment by testingbydev** [comment]
@FWAJL 
So we have to need  to allow user.role = 3 for project and task controller and other releted files.

**Comment by testingbydev** [comment]
@FWAJL 
i am enabling the project controller for user.role = 3 then if we have not any project in Session then it will redirect on proct select project then there are no project listed so we need to add a project but when we adding a project then there are error in "Call to a member function pm_id() on a non-object in <b>/opt/lampp/htdocs/FieldWorkAssistantMVC/Applications/PMTool/Controllers/ProjectController.php". so please help to solve this problem.

**Comment by FWAJL** [comment]
> @testingbydev said a day ago: 
> please tell about the process of how can we assign a project to FieldTechnician.

Brian answered this question here: https://github.com/FWAJL/FieldWorkAssistantMVC/issues/846#issuecomment-99182252

> @testingbydev said 23 hours ago:
> We Can't Find Any Session Project In Session if we have not any project in session then this is redirect Project list to select Project on project Controller but there are User Role 3 not Access the Project Controller So We have to need permitted project controller for user role 3 if we can't then how can we manage it in Mobile Controller and we cannot understand the process for a project and a task. please help to understanding the file structure and how can we listed task list in mobile controller.

As pointed out in [the first Step of this document](https://docs.google.com/document/d/1uSf9MTsVg0RzOapusLlkYn2noA2gZXOH86xFuZaZ10s/edit#heading=h.vnxfirhpqbrc), you need to use the MobileController, not the ProjectController nor the TaskController.
Also, in [the 4th Step of this document](https://docs.google.com/document/d/1uSf9MTsVg0RzOapusLlkYn2noA2gZXOH86xFuZaZ10s/edit#heading=h.4bb1btn1qu6a), we describe you that you need to build the session array to store the Task assigned to the technician. A technician can have one or many tasks. You need to list them with the filters described by Brian (https://github.com/FWAJL/FieldWorkAssistantMVC/issues/846#issuecomment-99182252). Then as stated in [the 4th Step of this document](https://docs.google.com/document/d/1uSf9MTsVg0RzOapusLlkYn2noA2gZXOH86xFuZaZ10s/edit#heading=h.4bb1btn1qu6a), the TaskHelper class holds the methods to fill the Task Session array.

> @testingbydev  said 10 hours ago:
> So we have to need to allow user.role = 3 for project and task controller and other releted files.

Yes, a FT is user.user_role_id = 3 and all the MobileController methods need to have this role set.
As pointed out in [the first Step of this document](https://docs.google.com/document/d/1uSf9MTsVg0RzOapusLlkYn2noA2gZXOH86xFuZaZ10s/edit#heading=h.vnxfirhpqbrc), you need to use the MobileController, not the ProjectController nor the TaskController.

> @testingbydev said 8 hours ago:
> i am enabling the project controller for user.role = 3 then if we have not any project in Session then it will redirect on proct select project then there are no project listed so we need to add a project but when we adding a project then there are error in "Call to a member function pm_id() on a non-object in /FieldWorkAssistantMVC/Applications/PMTool/Controllers/ProjectController.php". so please help to solve this problem.
Please tell about the process how can we listing task for field technician firstly which file need to be change.

Same as previous answer.

Thank you.

**Comment by testingbydev** [comment]
@FWAJL  @bsaiken

Jeremy we would definitely need your assistance in understanding the framework startup.
We have been able to understand following things in the MVC :
- Routing needs to be defined for user role.
- JS and CSS files structure.
- Resource XML files for adding key resources in local folder like mobile.en.xml .

Now what we need to understand from you is : 

- What is the use of Keys and constructor Within Resources/Enums/ViewVariables  ?
- What is the first step for listing the project and task of FT. where we need to write the first piece of code for this  , in helper or controller ?
- How can we manage at a time Project list and related task list within Mobile controller ? 
- Where we need to write the sql querries ?

your response to these questions would really help us in understanding your framework.

**Comment by testingbydev** [comment]
@bsaiken

we are not able to submit the partial code today because we are still not able to grab the MVC completely. your response to above questions may help us to move ahead.

**Comment by bsaiken** [comment]
@damirius 
Index page shall be:
 * Task Map (see Issue #848)
 * header (`menu show/hide` | `logo` | `Project > Task` (breadcrumb) | `FT Name`
 * left menu.  

Left menu items are:
a) Edit info
b) logout
c) communicate
d) view map
e) notes
f) Task list

**Comment by damirius** [comment]
@bsaiken

I've added few stuff on feature_mobile branch.

Add new menu options, moved logout/user edit in the menu, it will be shown only for FTs there.
Added breadcrumb for tasks in header together with FT name.
Landing page will be mobile/listTasks, where you can select the task by clicking on it. Only active tasks will be shown!
If there is only one active task in the list it will be automatically chosen and FT will be redirected on the map

**Comment by bsaiken** [comment]
@damirius 

On login I get a dead link.

![screen shot 2015-05-21 at 10 30 27 am](https://cloud.githubusercontent.com/assets/8228624/7754855/8667008c-ffa4-11e4-945b-f6b0c73ad2d9.png)

**Comment by damirius** [comment]
@bsaiken 

Do you have any active tasks related to this technician?

**Comment by bsaiken** [comment]
@damirius 
I tried starting from scratch.

I created a new Project and then tried creating a new Task for that Project and I get a similar error.
![screen shot 2015-05-21 at 10 50 20 am](https://cloud.githubusercontent.com/assets/8228624/7755191/43b7762e-ffa7-11e4-9514-14c098bd5796.png)

**Comment by damirius** [comment]
@bsaiken

Hm, this error I can't recreate. I'm starting to think that tasks aren't fetched here because of some problems with the db. Error is showing that foreach loop isn't getting proper argument, which should be an array of tasks. If there are no tasks, empty array should be passed. Most common reason for this is that db manager is getting some error.

**Comment by damirius** [comment]
@bsaiken

You can check out this video: https://drive.google.com/file/d/0B7Laiv0PjSvDSml3ZVRQQkFhUFE/view

**Comment by bsaiken** [comment]
@damirius 
I think you are correct.  When I was having trouble with some of Souvik's code, I had him send me the script for the database he was working with.  Then it worked.  So, send me your database.] and I will test with that.

The video looks perfect.

**Comment by damirius** [comment]
@bsaiken

You can find my local db backup in Database/Scripts/Online-backups/baiken_fwm_1_damir_20150521.sql

**Comment by bsaiken** [comment]
@damirius 

That's it.  Everything you have so far is working.  I think we can close this Issue.

Later today I can do some more unit testing.  Thanks!

**Comment by bsaiken** [comment]
@FWAJL @gsouvik 
This loads the page `user/showDetails`.

**Comment by bsaiken** [comment]
@damirius 
I think this is fixed.  If yes, please assign to Jeremie and Label for code review.
